c7n/resources/rds.py

```diff
@@ -1977,3 +1977,71 @@ class resource_type(TypeInfo):
         'describe': DescribeDBProxy,
         'config': ConfigSource
     }
+
+
+@filters.register('db-option-groups')
+class DbOptionGroups(ValueFilter):
+    """This filter describes RDS option groups for associated RDS instances.
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: rds-data-in-transit-encrypted
+            resource: aws.rds
+            filters:
+              - type: db-option-groups
+                key: OptionName
+                value: NATIVE_NETWORK_ENCRYPTION
+                op: eq
+    """
+
+    schema = type_schema('db-option-groups', rinherit=ValueFilter.schema)
+    schema_alias = False
+    permissions = ('rds:DescribeDBInstances', 'rds:DescribeOptionGroups', )
+
+    def handle_optiongroup_cache(self, client, paginator, option_groups):
+        pgcache = {}
+        cache = self.manager._cache
+
+        with cache:
+            for pg in option_groups:
+                cache_key = {
+                    'region': self.manager.config.region,
+                    'account_id': self.manager.config.account_id,
+                    'rds-pg': pg}
+                pg_values = cache.get(cache_key)
+                if pg_values is not None:
+                    pgcache[pg] = pg_values
+                    continue
+                option_list = list(itertools.chain(*[p['OptionGroupsList']
+                    for p in paginator.paginate(OptionGroupName=pg)]))
+
+                pgcache[pg] = {}
+                for option in option_list:
+                    if option['Options']:
+                        for p in option['Options']:
+                            pgcache[pg].update({'OptionName': p['OptionName']})
+                cache.save(cache_key, pgcache[pg])
+
+        return pgcache
+
+    def process(self, resources, event=None):
+        results = []
+        client = local_session(self.manager.session_factory).client('rds')
+        paginator = client.get_paginator('describe_option_groups')
+        option_groups = {db['OptionGroupMemberships'][0]['OptionGroupName']
+                        for db in resources}
```

Comment:
@sgungi-intuit this is what caused your unit test to randomly fail. You're using set comprehension here and there is no guarantee when it comes to ordering. `option_groups` could either be {'os-oracle-rds', 'default:postgres-13'} or {'default:postgres-13', 'os-oracle-rds'}. Depending on the ordering, it may or may not match with the recorded responses for DescribeOptionGroups, which then ended up causing to test to fail randomly.

The simple fix is to use list comprehension instead of set.